pkg/store/bucket.go

```diff
@@ -2255,7 +2251,11 @@ func (r *bucketChunkReader) preload() error {
 func (r *bucketChunkReader) loadChunks(ctx context.Context, offs []uint32, seq int, start, end uint32) error {
 	fetchBegin := time.Now()
 
-	b, err := r.block.readChunkRange(ctx, seq, int64(start), int64(end-start))
+	// Compute the byte ranges of chunks we actually need. The total read data may be bigger
+	// than required because of the partitioner.
+	chunkRanges := chunkOffsetsToByteRanges(offs, start)
```

Comment:
Comment on line 2301 is no longer valid (since now any chunk larger than 16000 bytes will cause additional read, not just last chunk), but the code still works.

I think it's worth extending the unit test to check for this scenario (chunks larger than 16000 bytes in the middle).